### PR TITLE
Add encoding detection (and pin export to UTF-8)

### PR DESCRIPTION
**Pull request by koppor**
This is a follow-up to https://github.com/JabRef/jabref/pull/8238.

This PR adds the feature to detect the encoding of a file. JabRef relies on the `% Encoding:` statement of a file. What if that statement is not present? JabRef 5.5 and 5.6 used "UTF-8"; versions before used the one in the user's preferences. Both approaches are not that reliable.

The PR also removes the feature of custom encoding for exporting - and inlines "UTF-8" settings (which were distributed throughout the code).

New general preferences:

grafik

It looked as follows before:

grafik
gi
- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/logic/importer/Importer.java [hunk 1]
```diff
@@ -97,6 +108,24 @@ public ParserResult importDatabase(Path filePath, Charset encoding) throws IOExc
         }
     }
 
+    protected static Charset getCharset(BufferedInputStream bufferedInputStream) {
+        Charset charset = StandardCharsets.UTF_8;
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/importer/Importer.java:112]: Shouldn't this fallback to the System default charset maybe?

Comment (koppor) [reply src/main/java/org/jabref/logic/importer/Importer.java:112]: Locally failed tests went up from 273 to 288 🙈

Comment (Siedlerchr) [reply src/main/java/org/jabref/logic/importer/Importer.java:112]: One of the problem I now encounter as well ( and I wonder why the tests previously didn't fail, is that the bibtex string resolving on import is making problems)

**Comment by Siedlerchr** [comment]
Only failing test is the StyleLoader test, but that apparently fails randomly

**Pull request merged**